### PR TITLE
Do not leak gh token

### PR DESCRIPTION
**Pull request by ZyX-I**

### CODE REVIEW

**Comment by ZyX-I** [comment]
This also limits number of attempts to push and checks whether pull --rebase succeeds in PRs.

**Comment by justinmk** [comment]
@ZyX-I If I'm not mistaken, it still leaks the token: https://travis-ci.org/neovim/bot-ci/jobs/228316769

Note: I didn't update with a new, valid token yet.

**Comment by ZyX-I** [comment]
@justinmk It is git leaking the token: it does not think that user is actually a password and not a user. Not my fault, use deploy keys. Though even if token was supplied as a password nothing would change:

```Shell
% git push 'https://git:oeasntuh@github.com/neovim/neovim'
remote: Invalid username or password.
fatal: Authentication failed for 'https://git:oeasntuh@github.com/neovim/neovim/'
```

**Comment by ZyX-I** [comment]
Note that you have automatic leak if GH is down or failed to connect:

```Shell
% git push 'https://git:oeasntuh@gthub.com/neovim/neovim'
fatal: unable to access 'https://git:oeasntuh@gthub.com/neovim/neovim/': Failed to connect to gthub.com port 443: В соединении отказано
```

so really, do not use `https://token@`. Or give up on error messages. Or redirect stderr and grep it out. For powerline/bot-ci I used deploy keys, as well as for populating https://github.com/ZyX-I/ZyX-I.github.io.

**Comment by justinmk** [comment]
I'll try `sed` for now.

**Comment by ZyX-I** [comment]
BTW, sync-lp-mirrors is already using deploy key. I do not think the key may be reused (that one is for launchpad) though, but code may.

File: ci/common/common.sh [hunk 1]
```diff
@@ -147,3 +153,23 @@ commit_subtree() {(
...
+    shift
+  else
+    set -u
+    : ${GH_TOKEN}
+  fi
+  for arg ; do
+    arg="${arg//%token/$GH_TOKEN}"
+    arg="${arg//%%/%}"
+    printf '%s\0' $arg
+  done | xargs -0 -x sh -c '"$@"' -
```
Comment (justinmk) [inline ci/common/common.sh]: @ZyX-I What is the purpose of `-` here? I am seeing `-: -n: command not found` in some cases.

Comment (ZyX-I) [inline ci/common/common.sh]: `sh -c 'echo "$@"' a b c` yields `b c`.

**Pull request merged**